src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,721 @@
+      if (field.required) require_fields.push_back(offset);
+      code_.SetValue("VALUENAME", name);
+      code_.SetValue("VALUETYPE", type);
+      code_.SetValue("OFFSET", NumToString(it - struct_def.fields.vec.begin()));
+      code_.SetValue("CONSTANT", field.value.constant);
+      code_ += "\tpublic static func add({{VALUENAME}}: \\";
+      if (IsScalar(field.value.type.base_type) &&
+          !IsBool(field.value.type.base_type)) {
+        std::string is_enum = IsEnum(field.value.type) ? ".rawValue" : "";
+        code_ += "{{VALUETYPE}}" + builder_string + "element: {{VALUENAME}}" +
+                 is_enum + ", def: {{CONSTANT}}, at: {{OFFSET}}) }";
+      } else if (IsBool(field.value.type.base_type)) {
+        code_.SetValue("VALUETYPE", "Bool");
+        code_.SetValue("CONSTANT",
+                       "0" == field.value.constant ? "false" : "true");
+        code_ +=
+            "{{VALUETYPE}}" + builder_string +
+            "condition: {{VALUENAME}}, def: {{CONSTANT}}, at: {{OFFSET}}) }";
+      } else {
+        std::string offset_type = field.value.type.base_type == BASE_TYPE_STRING
+                                      ? "Offset<String>"
+                                      : "Offset<UOffset>";
+        std::string reader_type =
+            IsStruct(field.value.type) && field.value.type.struct_def->fixed
+                ? "structOffset: {{OFFSET}}) }"
+                : "offset: {{VALUENAME}}, at: {{OFFSET}})  }";
+        code_ += offset_type + builder_string + reader_type;
+      }
+    }
+    code_ +=
+        "\tpublic static func end{{STRUCTNAME}}(_ fbb: FlatBufferBuilder, "
+        "start: "
+        "UOffset) -> Offset<UOffset> { let end = Offset<UOffset>(offset: "
+        "fbb.endTable(at: start))\\";
+    if (require_fields.capacity() != 0) {
+      std::string fields = "";
+      for (auto it = require_fields.begin(); it != require_fields.end(); ++it)
+        fields += *it + ", ";
+      code_.SetValue("FIELDS", fields.substr(0, fields.size() - 2));
+      code_ += "; fbb.require(table: end, fields: [{{FIELDS}}])\\";
+    }
+    code_ += "; return end }";
+
+    std::string spacing = "\t\t";
+    if (!struct_def.fixed && struct_def.has_key) {
+      code_.SetValue("VALUENAME", struct_def.name);
+      code_.SetValue("VOFFSET", NumToString(key_field->value.offset));
+
+      code_ +=
+          "\tpublic static func "
+          "sortVectorOf{{VALUENAME}}(offsets:[Offset<UOffset>], "
+          "_ fbb: FlatBufferBuilder) -> Offset<UOffset> {";
+      code_ += spacing + "var off = offsets";
+      code_ +=
+          spacing +
+          "off.sort { Table.compare(Table.offset(Int32($1.o), vOffset: "
+          "{{VOFFSET}}, fbb: fbb.buffer), Table.offset(Int32($0.o), vOffset: "
+          "{{VOFFSET}}, fbb: fbb.buffer), fbb: fbb.buffer) < 0 } ";
+      code_ += spacing + "return fbb.createVector(ofOffsets: off)";
+      code_ += "\t}";
+      GenLookup(*key_field);
```

Comment:
is key_field not-null here? what will happen if all fields are deprecated?

Reply:
this would be the same guard at the top